File: js/ui.js
/* jshint esversion: 6 */
/* global toBrowserH */

const getSansFont = () => {
   return toBrowserH(UI_SANS_TEXT_HEIGHT) + "px sans-serif";
};

const applyShadow = (ctx) => {
   ctx.shadowColor = "rgba(0, 0, 0, 0.2)";
   ctx.shadowBlur = toBrowserH(UI_SHADOW);
   ctx.shadowOffsetY = toBrowserH(UI_SHADOW);
};

const removeShadow = (ctx) => {
   ctx.shadowBlur = 0;
   ctx.shadowOffsetY = 0;
};

const Theme = {
   background: {
      r: 0.9,
      g: 0.9,
      b: 0.9
   },
   settings: {
      background: {
         r: 0.1,
         g: 0.1,
         b: 0.2
      }
   }
};

File: src/ui.rs
use crate::utils::{self, to_browser_h};

pub const UI_SHADOW: i32 = 20;
pub const UI_PADDING: i32 = 25;

pub const UI_SANS_TEXT_HEIGHT: i32 = 20;

/// The maximum alpha for 2D dialogs, notifications
pub const UI_MAX_ALPHA: f32 = 0.95;

fn get_sans_font() -> String {
   format!("{}px sans-serif", utils::to_browser_h(UI_SANS_TEXT_HEIGHT))
}

fn enable_shadow<CanvasContext>(ctx: CanvasContext) {
   ctx.shadow_color = "rgba(0, 0, 0, 0.2)";
   ctx.shadow_blur = to_browser_h(UI_SHADOW);
   ctx.shadow_offset_y = to_browser_h(UI_SHADOW);
}

fn remove_shadow<CanvasContext>(ctx: CanvasContext) {
   ctx.shadowBlur = 0;
   ctx.shadowOffsetY = 0;
}

struct Rgb(f32, f32, f32);

struct Theme {
    background: Rgb,
    settings_background: Rgb,
}

const THEME: Theme = Theme {
   background: (0.9, 0.9, 0.9),
   settings_background: (0.1, 0.1, 0.2)
};
